refund-crowdloan-contributors/refund-contributors.js

```diff
@@ -0,0 +1,104 @@
+const { ApiPromise, WsProvider } = require('@polkadot/api');
+const { Keyring } = require('@polkadot/keyring');
+const fs_promises = require('fs/promises');
+const fs = require('fs');
+const toml = require('toml');
+const jsonFile = require('./contributors');
+
+// Create a promise API instance of the passed in node address.
+async function createPromiseApi(nodeAddress) {
+    const wsProvider = new WsProvider(nodeAddress);
+    try {
+        const api = await new ApiPromise({ provider: wsProvider });
+        await api.isReady;
+        return api;
+    } catch (error) {
+        let message = "Failed to create client due to: " + error;
+        console.log(message);
+    }
+}
+
+function createBatchCalls(api, batch=[], para_id) {
+    let txs = []
+    for (const contributor of batch) {
+        let address = contributor['address'];
+        let tx = api.tx.crowdloan.withdraw(address, para_id);
+        txs.push(tx);
+    }
+
+    return txs;
+}
+
+
+async function queryBalances(api, batch=[]) {
+    let balances = []
+    for (const contributor of batch) {
+        let address = contributor['address'];
+        let { data: { free: free } } = await api.query.system.account(address);
+        balances.push(free);
+    }
+
+    return balances;
+}
+
+async function readProjectConfiguration(configPath) {
+    const content = await fs_promises.readFile(configPath, null, { encoding: 'utf-8' })
+    const config = toml.parse(content)
+    return config;
+}
+
+async function main() {
+    let configPath = "configuration.toml";
+    let config = await readProjectConfiguration(configPath);
+
+    const nodeAddress = config['node']['endpoint'];
+    const api = await createPromiseApi(nodeAddress);
+
+    let seed = config['node']['signer'];
+    const keyring = new Keyring({ type: 'sr25519' });
+    const sender = keyring.addFromUri(seed);
+
+    let balancesBefore = queryBalances(api, jsonFile);
+
+    let lengthJson = jsonFile.length;
+    let batchSize = config['node']['batch_size'];
+    let iterations = parseInt(lengthJson / batchSize);
```

Comment:
This doesn't fail if the number doesnt divide to int cleanly.
It'll truncate the iterations.
> If parseInt encounters a character that is not a numeral in the specified radix, it ignores it and all succeeding characters and returns the integer value parsed up to that point.

Could lead to an off-by-one problem where stuff from the last batch doesn't get broadcast.
Maybe explicitly fail this if they don't divide cleanly?

Reply:
This is handled in the loop with `i <= iterations`
The last batch can not full or empty.